common/transport/snmp/src/main/java/org/thingsboard/server/transport/snmp/service/SnmpTransportService.java

```diff
@@ -15,114 +15,75 @@
-import java.util.UUID;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
-import java.util.function.Consumer;
 
 @TbSnmpTransportComponent
 @Service
 @Slf4j
 public class SnmpTransportService implements TbTransportService {
     private final SnmpTransportContext snmpTransportContext;
+    private final TransportService transportService;
 
-    @Getter
-    private ExecutorService snmpCallbackExecutor;
-    @Getter
     private Snmp snmp;
     private ScheduledExecutorService pollingExecutor;
+    private ExecutorService snmpResponseProcessingExecutor;
 
-    public SnmpTransportService(@Lazy SnmpTransportContext snmpTransportContext) {
+    public SnmpTransportService(@Lazy SnmpTransportContext snmpTransportContext,
+                                TransportService transportService) {
         this.snmpTransportContext = snmpTransportContext;
+        this.transportService = transportService;
     }
 
-    //    @PostConstruct
-    private void init() {
-        log.info("Starting SNMP transport...");
+    @PostConstruct
+    private void init() throws IOException {
+        log.info("Initializing SNMP transport service");
+
         pollingExecutor = Executors.newScheduledThreadPool(1, ThingsBoardThreadFactory.forName("snmp-polling"));
         //TODO: Set parallelism value in the config
-        snmpCallbackExecutor = Executors.newWorkStealingPool(20);
+        snmpResponseProcessingExecutor = Executors.newWorkStealingPool(20);
 
         initializeSnmp();
 
-        log.info("SNMP transport started!");
-    }
-
-    @PreDestroy
-    public void shutdown() {
-        log.info("Stopping SNMP transport!");
-        if (pollingExecutor != null) {
-            pollingExecutor.shutdownNow();
-        }
-        if (snmpCallbackExecutor != null) {
-            snmpCallbackExecutor.shutdownNow();
-        }
-        if (snmp != null) {
-            try {
-                snmp.close();
-            } catch (IOException e) {
-                log.error(e.getMessage(), e);
-            }
-        }
-        log.info("SNMP transport stopped!");
-    }
-
-    @EventListener(ApplicationReadyEvent.class)
-    @Order(value = 10)
-    public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
-        log.info("Received application ready event. Starting SNMP polling.");
-//        startPolling();
+        log.info("SNMP transport service initialized");
     }
 
-    private void initializeSnmp() {
-        try {
-            this.snmp = new Snmp(new DefaultUdpTransportMapping());
-            this.snmp.listen();
-        } catch (IOException e) {
-            //TODO: what should be done if transport wasn't initialized?
-            log.error(e.getMessage(), e);
-        }
+    private void initializeSnmp() throws IOException {
+        snmp = new Snmp(new DefaultUdpTransportMapping());
```

Comment:
Maybe someone would like to support SNMP over TCP? Let's add TODO to make this configurable.